From a720757fe70d08e3023283242b1af9eb4364bdeb Mon Sep 17 00:00:00 2001
From: Wadim Egorov <w.egorov@phytec.de>
Date: Fri, 31 Jul 2020 07:35:07 +0000
Subject: [PATCH] FORUPSTREAM: phycore-rk3288: Remove PMIC related stuff from
 SPL

We can remove the power stuff at the SPL stage from our board. The SOM
was redesigned and is equipped with an STM8 connected to the RK818.
The required setup we did before in the SPL is now done by the ST
controller.  know there are only a few SOMs without the STM8 out in
the wild. So if I remove it it will affect only a few people who probably
already have both boards.

Signed-off-by: Wadim Egorov <w.egorov@phytec.de>
---
 board/phytec/phycore_rk3288/phycore-rk3288.c | 42 ----------------------------
 configs/phycore-rk3288_defconfig             |  2 --
 2 files changed, 44 deletions(-)

diff --git a/board/phytec/phycore_rk3288/phycore-rk3288.c b/board/phytec/phycore_rk3288/phycore-rk3288.c
index 039ed0f..e9c514f 100644
--- a/board/phytec/phycore_rk3288/phycore-rk3288.c
+++ b/board/phytec/phycore_rk3288/phycore-rk3288.c
@@ -15,8 +15,6 @@
 #include <i2c_eeprom.h>
 #include <netdev.h>
 #include "som.h"
-#include <power/regulator.h>
-#include <power/rk8xx_pmic.h>
 
 static int valid_rk3288_som(struct rk3288_som *som)
 {
@@ -74,46 +72,6 @@ int rk3288_board_late_init(void)
 	return 0;
 }
 
-#ifdef CONFIG_SPL_BUILD
-#if !defined(CONFIG_SPL_OF_PLATDATA)
-static int phycore_init(void)
-{
-	struct udevice *pmic;
-	int ret;
-
-	ret = uclass_first_device_err(UCLASS_PMIC, &pmic);
-	if (ret)
-		return ret;
-
-#if defined(CONFIG_SPL_POWER_SUPPORT)
-	/* Increase USB input current to 2A */
-	ret = rk818_spl_configure_usb_input_current(pmic, 2000);
-	if (ret)
-		return ret;
-
-	/* Close charger when USB lower then 3.26V */
-	ret = rk818_spl_configure_usb_chrg_shutdown(pmic, 3260000);
-	if (ret)
-		return ret;
-#endif
-
-	return 0;
-}
-#endif
-
 void spl_board_init(void)
 {
-#if !defined(CONFIG_SPL_OF_PLATDATA)
-	int ret;
-
-	if (of_machine_is_compatible("phytec,rk3288-phycore-som")) {
-		ret = phycore_init();
-		if (ret) {
-			debug("Failed to set up phycore power settings: %d\n",
-			      ret);
-			return;
-		}
-	}
-#endif
 }
-#endif
diff --git a/configs/phycore-rk3288_defconfig b/configs/phycore-rk3288_defconfig
index 4c48d42..44bf439 100644
--- a/configs/phycore-rk3288_defconfig
+++ b/configs/phycore-rk3288_defconfig
@@ -20,8 +20,6 @@ CONFIG_DEFAULT_FDT_FILE="rk3288-phycore-rdk.dtb"
 CONFIG_DISPLAY_BOARDINFO_LATE=y
 CONFIG_SPL_STACK_R=y
 CONFIG_SPL_STACK_R_MALLOC_SIMPLE_LEN=0x2000
-CONFIG_SPL_I2C_SUPPORT=y
-CONFIG_SPL_POWER_SUPPORT=y
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_GPT=y
 CONFIG_CMD_I2C=y
-- 
2.7.4

